com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/linking/LinkingService.java

```diff
@@ -82,13 +84,22 @@ private void registerNamedType(final EObject context, final QualifiedName name,
    *          the lookup type, may be {@code null}
    */
   public void importObject(final EObject context, final EObject target, final EClass type) {
-    final IResourceDescriptions resourceDescriptions = provider.getResourceDescriptions(context.eResource());
-    Iterator<IEObjectDescription> exports = resourceDescriptions.getExportedObjectsByObject(target).iterator();
-    if (exports.hasNext()) {
-      QualifiedName targetName = exports.next().getName();
-      if (targetName != null && !targetName.isEmpty()) {
-        registerNamedType(context, targetName.toLowerCase(), type); // NOPMD targetName not a String!
+    Resource eResource = target.eResource();
+    QualifiedName targetName = null;
+    if (eResource instanceof LazyLinkingResource2) {
+      IQualifiedNameProvider nameProvider = ((LazyLinkingResource2) eResource).getService(IQualifiedNameProvider.class);
+      if (nameProvider != null) {
+        targetName = nameProvider.getFullyQualifiedName(target);
       }
+    } else {
+      final IResourceDescriptions resourceDescriptions = provider.getResourceDescriptions(context.eResource());
+      Iterator<IEObjectDescription> exports = resourceDescriptions.getExportedObjectsByObject(target).iterator();
+      if (exports.hasNext()) {
+        targetName = exports.next().getName();
+      }

```

Comment:
I think this code will never be executed, and it is indeed never executed in a FULL build with ASMD, I did not removed it because I am not sure if the same holds for ABMD/AMI-WS, even though I think it should. What do you think, should I remove it?

Reply:
In general we may have resources that are not LazyLinkingResource2 ones, so let's keep it